src/core/viz/expressions/Animation.js

```diff
@@ -93,7 +96,46 @@ export class Animation extends BaseExpression {
         return !this.paused;
     }
 
+    _dataReady() {
+        if (waitingForLayer.has(this)) {
+            waitingForLayer.delete(this);
+            waitingForOthers.add(this);
+        }
+        if (waitingForOthers.has(this)) {
+            waitingForLayer = new Set([...waitingForLayer].filter(expr=>{
+                while (expr.parent){
+                    expr = expr.parent;
+                }
+                if (expr._getRootExpressions){
+                    // The animation hasn't been removed from the viz
+                    return true;
+                }
+                return false;
+            }));
+            if (waitingForLayer.size > 0) {
+                return;
+            }
+            [...waitingForOthers.values()].map(anim => {
+                if (anim._paused === 'default') {
+                    anim._paused = false;
+                }
+            });
+            waitingForOthers.clear();
+        }
+    }
+
+    _postShaderCompile(program, gl) {
+        waitingForLayer.add(this);
+        this._paused = 'default';
```

Comment:
I don't find `default`quite clear... maybe `'uninitialized'`? `'waiting-for-start'`? `'waiting'`?

Reply:
Which are the values could have `paused`? `true`, `false` and 'default? Maybe, instead, I would use a different property `this._initialized` instead, but I have to take a deeper look.